packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -1053,7 +1053,7 @@ class PopupMenuButtonState<T> extends State<PopupMenuButton<T>> {
     final RelativeRect position = RelativeRect.fromRect(
       Rect.fromPoints(
         button.localToGlobal(widget.offset, ancestor: overlay),
-        button.localToGlobal(button.size.bottomRight(Offset.zero), ancestor: overlay),
+        button.localToGlobal(button.size.bottomRight(Offset.zero) + widget.offset, ancestor: overlay),
```

Comment:
I'm pretty sure we also need to move the `bottomRight`, otherwise, if the `offset` bigger than the button size, it has a wrong offset, which was tested at the following test.